File: backend/app/data/project.py
"""项目 mock 数据"""

from datetime import datetime, timedelta

PROJECTS = [
    {
        "id": f"proj-{i}",
        "title": ["LKM 论坛优化", "理科竞赛题库系统", "匿名树洞重构", "StarHope AI 辅助"][i],
        "description": f"项目 {i+1} 的详细描述",
        "cover_image": "",
        "author_id": "user-1",
        "author_name": "项目发起人",
        "status": ["recruiting", "active", "completed", "archived"][i % 4],
        "member_count": i * 3 + 2,
        "created_at": (datetime(2026, 6, 1) + timedelta(days=i * 30)).isoformat(),
    }
    for i in range(4)
]

File: src/lib/api/modules/project.ts
import { get } from "../../http/client";
import type { PaginatedResponse } from "../types";

export type { PaginatedResponse } from "../types";

export interface Project {
  id: string;
  title: string;
  description: string;
  coverImage: string;
  authorId: string;
  authorName: string;
  status: string;
  memberCount: number;
  createdAt: string;
}

export const projectApi = {
  getProjects: (page = 1, limit = 20) =>
    get<PaginatedResponse<Project>>("/api/projects", { page, limit }),

  getProject: (id: string) => get<Project>(`/api/projects/${id}`),
};
